catalog/task/collect-data/0.1/collect-data.yaml

```diff
@@ -53,32 +53,43 @@ spec:
       script: |
         #!/usr/bin/env sh
 
-        if ! release_result=$(get-resource "release" "${RELEASE}" "{.spec.extraData}"); then
+        if ! get-resource "release" "${RELEASE}" > "$(workspaces.data.path)/release.json"; then
```

Comment:
So here we basically just switched the order - first we load the whole resources and then later we just get the `extraData` field for each. I think this is better. But effectively this shouldn't change the functionality, right? Because if the resource doesn't even exist, even previously it would fail on this, right? Or what am I missing?

Reply:
I just switched the order because I want to fail early if I am going to fail (if a CR is missing), instead of computing some jsons first then failing

Reply:
OK, sure, but you agree it would fail anyway? The commit title suggests that only after this change it will now fail if a CR is missing.

Reply:
It would fail anyway without the reordering, yes. It will not fail without this PR because there is no `exit 1` after checking the return code of the `get-resource > workspace` lines

Reply:
>  It will not fail without this PR because there is no exit 1 after checking the return code of the get-resource > workspace lines

But there is `exit 1` here at the top. For example, if the `release` resource is missing, this `if` will trigger and it will `exit 1`.
```sh
if ! release_result=$(get-resource "release" "${RELEASE}" "{.spec.extraData}");
```
Is that not the case?

Reply:
Right. But what if the `snapshot` does not exist? We don't do anything with `extraData` from that. So I was consistent and changed to the check for all the retrievals to the workspace

Reply:
OK, fair enough. So can you please at least explain this in the commit message? That in most cases it would already fail, but for example if `snapshot` was missing, it would previously not fail?

Reply:
Yeah, I will wait to see what happens with the `set -e` thing then adjust accordingly

Reply:
@mmalina fixed

Reply:
cool, thanks